src/library_wasi.js

```diff
@@ -543,16 +535,16 @@ var WasiLibrary = {
 #if SYSCALLS_REQUIRE_FILESYSTEM
     var stream = SYSCALLS.getStreamFromFD(fd);
 #if ASYNCIFY
-    return Asyncify.handleSleep(function(wakeUp) {
+    return Asyncify.handleSleep((wakeUp) => {
       var mount = stream.node.mount;
       if (!mount.type.syncfs) {
         // We write directly to the file system, so there's nothing to do here.
         wakeUp(0);
         return;
       }
-      mount.type.syncfs(mount, false, function(err) {
+      mount.type.syncfs(mount, false, (err) => {
         if (err) {
-          wakeUp(function() { return {{{ cDefs.EIO }}} });
+          wakeUp({{{ cDefs.EIO }}});
```

Comment:
Is this a bugfix?

Reply:
Ah, yes.  Maybe I should submit this separately?

Reply:
That PR lgtm and this one lgtm as well to land after it.